### PR TITLE
Proper handling of multiple concurrent requests for the same certificate

### PR DESCRIPTION
**Pull request by ArnyminerZ**
I think this should be enough. If the error is thrown it's because a response has already been given, so we can't do anything else than just not sending anything more.

### CODE REVIEW

**Comment by ArnyminerZ** [comment]
I will take a further look this afternoon

**Comment by ArnyminerZ** [comment]
I haven't been able to reproduce, and there's nothing super obvious... I've synchronized `pendingDecisions` just in case there's a threading issue or something. But not sure what the problem is

File: lib/src/main/java/at/bitfire/cert4android/UserDecisionRegistry.kt [hunk 1]
```diff
@@ -108,10 +108,15 @@ class UserDecisionRegistry private constructor(
...
-                iter.next().resume(trusted)
-                iter.remove()
+        synchronized(pendingDecisions) {
+            pendingDecisions[cert]?.iterator()?.let { iter ->
+                while (iter.hasNext()) {
+                    try {
+                        iter.next().resume(trusted)
+                    } finally {
+                        iter.remove()
+                    }
```
Comment (rfc2822) [inline lib/src/main/java/at/bitfire/cert4android/UserDecisionRegistry.kt:118]: Hm but try / finally without catch would still throw the Exception, wouldn't it?

Comment (ArnyminerZ) [reply lib/src/main/java/at/bitfire/cert4android/UserDecisionRegistry.kt:118]: Yeah, that's true, I'll add a simple catch

File: lib/src/main/java/at/bitfire/cert4android/UserDecisionRegistry.kt [hunk 2]
```diff
@@ -108,10 +108,17 @@ class UserDecisionRegistry private constructor(
...
-            while (iter.hasNext()) {
-                iter.next().resume(trusted)
-                iter.remove()
+        synchronized(pendingDecisions) {
+            pendingDecisions[cert]?.iterator()?.let { iter ->
+                while (iter.hasNext()) {
+                    try {
+                        iter.next().resume(trusted)
+                    } catch(_: IllegalStateException) {
+                        Cert4Android.log.warning("Tried to resume an already resumed continuation.")
```
Comment (rfc2822) [inline lib/src/main/java/at/bitfire/cert4android/UserDecisionRegistry.kt:117]: Hm I'm not that happy with a "blind" catch when we don't know what's it actually for. I think we should put more efforts in finding out the cause of the problem and make sure that the continuation can't be resumed twice. If we can't guarantee that, continuation may not be the right way to do it and we should use `CompleteableDeferred` or whatever else.

**Comment by rfc2822** [comment]
Should work now.

@ArnyminerZ Can you please try the most important cases (and the one that crashed) to see whether they work for you? As a double-check.

**Pull request merged**